src/main/java/seedu/clinic/logic/commands/FindCommand.java

```diff
@@ -1,42 +1,65 @@
 package seedu.clinic.logic.commands;
 
+import static java.util.Objects.nonNull;
 import static java.util.Objects.requireNonNull;
 
 import seedu.clinic.commons.core.Messages;
 import seedu.clinic.model.Model;
-import seedu.clinic.model.attribute.NameContainsKeywordsPredicateForSupplier;
+import seedu.clinic.model.supplier.SupplierProductsContainKeywordsPredicate;
+import seedu.clinic.model.warehouse.WarehouseProductsContainKeywordsPredicate;
 
 /**
- * Finds and lists all suppliers/warehouses in the CLI-nic app whose name contains any of the argument keywords.
+ * Finds and lists all suppliers/warehouses in the CLI-nic app that sell/hold products matching any of the argument
+ * keywords.
  * Keyword matching is case insensitive.
+ * Keyword only matches whole word e.g. band will match "band aid" but not "bandaid".
  */
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all suppliers whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all suppliers/warehouses that sell/hold products"
+            + "matching the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: TYPE KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " supplier panadol"
+            + "Example: " + COMMAND_WORD + " warehouse face mask";
 
-    private final NameContainsKeywordsPredicateForSupplier predicate;
+    private final SupplierProductsContainKeywordsPredicate supplierPredicate;
+    private final WarehouseProductsContainKeywordsPredicate warehousePredicate;
 
-    public FindCommand(NameContainsKeywordsPredicateForSupplier predicate) {
-        this.predicate = predicate;
+    /**
+     * Constructs a new FindCommand object.
+     *
+     * @param supplierPredicate takes in the predicate used to filter a supplier's products.
+     * @param warehousePredicate takes in the predicate used to filter a warehouse's products.
+     */
+    public FindCommand(SupplierProductsContainKeywordsPredicate supplierPredicate,
+                       WarehouseProductsContainKeywordsPredicate warehousePredicate) {
+        this.supplierPredicate = supplierPredicate;
+        this.warehousePredicate = warehousePredicate;
     }
```

Comment:
Is it possible to have both predicates in the constructor? Maybe we need 2 overloaded constructors, or we can test on cases where both predicates are added?